pkg/apiserver/rest/usecase/authentication.go

```diff
@@ -287,8 +290,12 @@ func (a *authenticationUsecaseImpl) GetLoginType(ctx context.Context) (*apisv1.G
 	if err != nil {
 		return nil, err
 	}
+	loginType := sysInfo.LoginType
+	if loginType == "" {
```

Comment:
The loginType should be set when systemInfo is initiliazed so we don't need to check here second time.

Reply:
Yes, if the user upgrades from 1.2, the login type will be empty.